x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/IdentifierBuilder.java

```diff
@@ -27,6 +27,10 @@ public String visitQualifiedName(QualifiedNameContext ctx) {
     }
 
     private static String unquoteIdentifier(String identifier) {
-        return identifier.replace("`", "");
+        String newIdentifier = identifier;
+        if (identifier.charAt(0) == '`') {
+            newIdentifier = identifier.substring(1, identifier.length() - 1);
+        }
+        return newIdentifier.replace("``", "`");
```

Comment:
Why is there a `newIdentifier` and the existing identifier is not reused? 
Also shouldn't the condition check that the identifiers starts and stops with a backtick?
Please add a test for the following cases:
`bar
``bar`
```bar`
bar`
bar``

Reply:
I can reuse it, it's just personal pref to not change the passed arg.
Not necessary because if it's correctly parsed from the parser it always has the ending backquote, we use the same checks for the `unquoteString` method.

Will add more tests